ingestion/functions/parsing/colombia/colombia.py

```diff
@@ -43,36 +41,56 @@ def convert_gender(raw_gender):
 def convert_location(raw_entry):
     query_terms = [
         term for term in [
-            raw_entry.get("Ciudad de ubicación", ""),
-            raw_entry.get("Departamento o Distrito ", ""),
+            raw_entry.get("Nombre municipio", ""),
+            raw_entry.get("Nombre departamento", ""),
             "Colombia"]
-        if term]
+        if term != '']
 
     return {"query": ", ".join(query_terms)}
 
 
-def get_ethnicity(raw_entry):
-    ethnicity_map = {'Otro': 'Other',
-                     'Negro': 'Black',
-                     'Indigeno': 'Indigenous'}
-    if ethnicity_map.get(raw_entry['Pertenencia etnica'], None):
-        if raw_entry['Pertenencia etnica'] == 'Indígena':
-            if raw_entry['Nombre grupo etnico'] != '' and raw_entry['Nombre grupo etnico'] != 'Sin Comunidad' and raw_entry['Nombre grupo etnico'] != 'Sin definir':
-                return f"Indigenous, {raw_entry['Nombre grupo etnico']}"
-            return "Indigenous"
-        return ethnicity_map.get(raw_entry['Pertenencia etnica'], None)
-    return None
+def convert_demographics(entry):
+    '''
+    This takes a whole row, and returns Age, Gender and Ethnicity in a dictionary.
+    Age is given as an int, but the adjacent field, 'Age Measurement Unit', determines what this int represents.
+    1 = Years; 2 = Months; 3 = Days
+    '''
 
+    ethnicity_map = {'1': 'Indigenous',
+                     '2': 'ROM',
+                     '3': 'Raizal',
+                     '4': 'Palenquero',
+                     '5': 'Black',
+                     '6': 'Other'}
 
-def convert_demographics(age: str, sex: str):
     demo = {}
-    if age:
-        demo["ageRange"] = {
-            "start": float(age),
-            "end": float(age)
-        }
-    if sex:
-        demo["gender"] = convert_gender(sex)
+    if entry['Edad']:
```

Comment:
As before, this will still raise `KeyError` if the entry is not in the dictionary. Please change these to the safer `if 'Edad' in entry:`.

Reply:
@fojackson8 and I discussed this, we think it's better to fail if the columns aren't present because it means the team gets informed that the data cannot be parsed, rather than the parser "succeeding" to create no cases or use incomplete data.